asapdiscovery-modeling/asapdiscovery/modeling/scripts/slice_fragalysis_structures_with_mdtraj.py

```diff
@@ -0,0 +1,131 @@
+"""
+This script slices the structures from the fragalysis database into the active site and full protein PDB files,
+as well as the corresponding numpy arrays of the coordinates of the atoms in the active site and full protein.
+"""
+import argparse
+import logging
+import multiprocessing as mp
+from datetime import datetime
+from pathlib import Path
+
+import mdtraj as md
+import numpy as np
+from asapdiscovery.data.utils import check_filelist_has_elements
+
+
+################################################################################
+def get_args():
+    parser = argparse.ArgumentParser(description="")
+
+    # Input arguments
+    parser.add_argument(
+        "-d",
+        "--structure_dir",
+        required=True,
+        help="Path to structure directory.",
+    )
+    parser.add_argument(
+        "-o",
+        "--output_dir",
+        required=True,
+        help="Path to output directory.",
+    )
+
+    parser.add_argument(
+        "-l",
+        "--log_file",
+        type=str,
+        default="save_combined_frag_structures.log",
+        help="Log file",
+    )
+
+    # Performance arguments
+    parser.add_argument(
+        "-n",
+        "--num_cores",
+        type=int,
+        default=1,
+        help="Number of concurrent processes to run.",
+    )
+
+    return parser.parse_args()
+
+
+def analyze_mp(fn, out_dir):
+    full_protein_selection = "not element H and (chainid 0 or chainid 2)"
+    active_site_string = "not element H and (chainid 0 or chainid 2) and (residue 140 to 145 or residue 163 or residue 172 or residue 25 to 27 or residue 41 or residue 49 or residue 54 or residue 165 to 168 or residue 189 to 192)"
+    output_name = fn.stem
+
+    # Prepare logger
+    handler = logging.FileHandler(out_dir / f"{output_name}-log.txt", mode="w")
```

Comment:
Use `FileLogger` if it suits?

Reply:
done https://github.com/choderalab/asapdiscovery/pull/105/commits/675340cf108690c063443de9ce861476c87f7485